pkg/logger/logger.go

```diff
@@ -2,34 +2,186 @@ package logger
 
 import (
 	"fmt"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
 
-	"github.com/fatih/color"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/kubectl-ces-plugin/cmd/plugin/cli/util"
+)
+
+// LogLevelKey identifies
+const LogLevelKey = "log-level"
+
+// CLI log level strings
+const (
+	LogLevelError LogLevel = "error"
+	LogLevelWarn  LogLevel = "warn"
+	LogLevelInfo  LogLevel = "info"
+	LogLevelDebug LogLevel = "debug"
+)
+
+// logrus log levels
+const (
+	errorLevel int = iota
+	warningLevel
+	infoLevel
+	debugLevel
 )
 
-type Logger struct {
+var (
+	loggerInstance *NamedLogger
+	formatter      = createFormatter()
+)
+
+func init() {
+	initLoggerWithThrowAway()
+}
+
+func initLoggerWithThrowAway() {
+	// the loggers-to-be-used are instantiated later on when the CLI log level flags have been parsed
+	log := logrus.New()
+	namedLog := &NamedLogger{logger: log, name: "throwAway"}
+	loggerInstance = namedLog
+	core.GetLogger = func() core.Logger {
+		return namedLog
+	}
+}
+
+func GetInstance() *NamedLogger {
+	return loggerInstance
+}
+
+func createFormatter() Formatter {
+	formatter := Formatter{
+		HideKeys:        true,
+		FieldsOrder:     []string{"component", "category"},
+		TimestampFormat: time.RFC3339,
+		NoColors:        true,
+		CallerFirst:     true,
+		CustomCallerFormatter: func(frame *runtime.Frame) string {
+			file := strings.Split(frame.File, "/")
+			function := strings.Split(frame.Function, ".")
+			return file[len(file)-1] + ":" + function[len(function)-1] + ":" + strconv.Itoa(frame.Line) + " "
+		},
+	}
+
+	return formatter
+}
+
+func getLogLevelFromEnv() logrus.Level {
+	logLevelRef := viper.GetViper().Get(util.CliTransportLogLevel)
+	logLevel := *(logLevelRef).(*LogLevel)
```

Comment:
🚫 **[checkstyle]** <forcetypeassert> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type assertion must be checked